ios/RCTMapboxGL/RCTMapboxGLConversions.m

```diff
@@ -54,48 +60,32 @@
         
-        RCTMGLAnnotation *pin = [[RCTMGLAnnotation alloc] initWithLocationRightCallout:CLLocationCoordinate2DMake(lat, lng) title:title subtitle:subtitle id:id rightCalloutAccessory:imageButton];
-        
-        if ([annotationObject valueForKey:@"annotationImage"]) {
-            NSObject *annotationImage = [annotationObject valueForKey:@"annotationImage"];
-            NSString *annotationImageURL = [annotationImage valueForKey:@"url"];
-            CGFloat height = (CGFloat)[[annotationImage valueForKey:@"height"] floatValue];
-            CGFloat width = (CGFloat)[[annotationImage valueForKey:@"width"] floatValue];
-            if (!height || !width) {
-                RCTLogError(@"Height and width for image required");
-                return nil;
-            }
-            CGSize annotationImageSize =  CGSizeMake(width, height);
-            pin.annotationImageURL = annotationImageURL;
-            pin.annotationImageSize = annotationImageSize;
-        }
-        
-        return pin;
+        point = [[RCTMGLAnnotation alloc] initWithLocationRightCallout:CLLocationCoordinate2DMake(lat, lng) title:title subtitle:subtitle id:id rightCalloutAccessory:imageButton];
         
     } else {
         
         NSArray *coordinate = [RCTConvert NSArray:[annotationObject valueForKey:@"coordinates"]];
         CLLocationDegrees lat = [coordinate[0] doubleValue];
         CLLocationDegrees lng = [coordinate[1] doubleValue];
         
-        RCTMGLAnnotation *point = [[RCTMGLAnnotation alloc] initWithLocation:CLLocationCoordinate2DMake(lat, lng) title:title subtitle:subtitle id:id];
-        
-        if ([annotationObject valueForKey:@"annotationImage"]) {
-            NSObject *annotationImage = [annotationObject valueForKey:@"annotationImage"];
-            NSString *annotationImageURL = [annotationImage valueForKey:@"url"];
-            CGFloat height = (CGFloat)[[annotationImage valueForKey:@"height"] floatValue];
-            CGFloat width = (CGFloat)[[annotationImage valueForKey:@"width"] floatValue];
-            if (!height || !width) {
-                RCTLogError(@"Height and width for image required");
-                return nil;
-            }
-            CGSize annotationImageSize =  CGSizeMake(width, height);
-            point.annotationImageURL = annotationImageURL;
-            point.annotationImageSize = annotationImageSize;
+        point = [[RCTMGLAnnotation alloc] initWithLocation:CLLocationCoordinate2DMake(lat, lng) title:title subtitle:subtitle id:id];
+    }
+    
+    if ([annotationObject valueForKey:@"annotationImage"]) {
+        NSDictionary *annotationImage = [annotationObject valueForKey:@"annotationImage"];
+        NSDictionary *imageSource = (NSDictionary*)annotationImage[@"source"];
+        CGFloat height = (CGFloat)[[annotationImage valueForKey:@"height"] floatValue];
+        CGFloat width = (CGFloat)[[annotationImage valueForKey:@"width"] floatValue];
```

Comment:
`CGFloat` is actually a typedef for `double`, so you can avoid the cast by calling `-doubleValue` instead of `-floatValue`.